src/Microsoft.AspNetCore.Hosting.Abstractions/IWebHostBuilder.cs

```diff
@@ -47,6 +48,12 @@ public interface IWebHostBuilder
         IWebHostBuilder UseSetting(string key, string value);
 
         /// <summary>
+        /// Indicate that the host should listen on the URLs configured on the <see cref="IWebHostBuilder"/> instead of those configured on the <see cref="IServer"/>.
+        /// </summary>
+        /// <returns>The <see cref="IWebHostBuilder"/>.</returns>
+        IWebHostBuilder PreferHostingUrls();
```

Comment:
Only if we flow this through configuration which exposes it to things like environment variables, ...